File: c++/ARRAY/largest_three.cpp
// #include<iostream>
// using namespace std;
// int arr[100], i, n, first, second, third;
// int main()
// {
//     cout<<"Enter number:";
//     cin>>n;
//     if(n<3)
//         cout<<"Invalid output!";
//     for(i=0; i<n; i++)
//         cin>>arr[i];
//     first = arr[0];
//     for(i=0; i<n; i++)
//     {
//         if(arr[i]>first)
//         {
//             third=second;
//             second=first;
//             first=arr[i];
//         }
//         else if(arr[i]>second)
//         {
//             third= second;
//             second=arr[i];
//         }
//         else if(arr[i]>third)
//             third=arr[i];
//     }
//      cout<<"Three largest number:"<<first<<"  "<<second<<"  "<<third;    
// }

#include<iostream>
using namespace std;
int arr[100], i, n, first, second, third;
int three(int arr[100], int n)
{
    first = arr[0];
    for(i=0; i<n; i++)
    {
        if(arr[i]>first)
        {
            third=second;
            second=first;
            first=arr[i];
        }
        else if(arr[i]>second)
        {
            third= second;
            second=arr[i];
        }
        else if(arr[i]>third)
            third=arr[i];
    }
     cout<<"Three largest number:"<<first<<"  "<<second<<"  "<<third;    
}

int main()
{
    cout<<"enter number of elements:";
    cin>>n;
    if(n<3)
        cout<<"Invalid output!";
    // first = arr[0]; 
    for(i=0; i<n; i++)
        cin>>arr[i];
    three(arr,3);
    return 0;
}

File: java/array/Largest_three.java
// LARGEST THREE
package array;
public class Largest_three 
{
    public static void main(String[] args)
    {
        int[] arr = {24, 701, 202, 983, 345};
        int first = Integer.MIN_VALUE, second = Integer.MIN_VALUE, third = Integer.MIN_VALUE ;
        for(int i=0; i<arr.length; i++){
            if(arr[i]>first){
                third = second;
                second = first;
                first = arr[i];
            }
            else if(arr[i]>second){
                third = second;
                second =arr[i];
            }
            else if(arr[i]>third){
                third = arr[i];
            }
        }
        System.out.println("First largest element is:"+first);
        System.out.println("Second largest element is:"+second);
        System.out.println("Third largest element is:"+third);
    }
}


// SMALLEST THREE


// public class Largest_three 
// {
//     public static void main(String[] args)
//     {
//         int[] arr = {24, 701, 202, 983, 345};
//         int first = Integer.MAX_VALUE, second = Integer.MAX_VALUE, third = Integer.MAX_VALUE ;
//         for(int i=0; i<arr.length; i++){
//             if(arr[i]<first){
//                 third = second;
//                 second = first;
//                 first = arr[i];
//             }
//             else if(arr[i]<second){
//                 third = second;
//                 second =arr[i];
//             }
//             else if(arr[i]<third){
//                 third = arr[i];
//             }
//         }
//         System.out.println("First smallest element is:"+first);
//         System.out.println("Second smallest element is:"+second);
//         System.out.println("Third smallest element is:"+third);
//     }
// }
